android/KMEA/app/src/main/java/com/keyman/engine/KMManager.java

```diff
@@ -2217,21 +2217,49 @@ public static int getBannerHeight(Context context) {
     return bannerHeight;
   }
 
+  // Overloaded method with an additional orientation parameter
   public static int getKeyboardHeight(Context context) {
+    int orientation = getOrientation(context);
     int defaultHeight = (int) context.getResources().getDimension(R.dimen.keyboard_height);
     SharedPreferences prefs = context.getSharedPreferences(context.getString(R.string.kma_prefs_name), Context.MODE_PRIVATE);
-
-    int orientation = getOrientation(context);
     if (orientation == Configuration.ORIENTATION_PORTRAIT) {
-      return prefs.getInt(KMManager.KMKey_KeyboardHeightPortrait, defaultHeight);
+        return prefs.getInt(KMManager.KMKey_KeyboardHeightPortrait, defaultHeight);
     } else if (orientation == Configuration.ORIENTATION_LANDSCAPE) {
-      return prefs.getInt(KMManager.KMKey_KeyboardHeightLandscape, defaultHeight);
+        return prefs.getInt(KMManager.KMKey_KeyboardHeightLandscape, defaultHeight);
     }
-
     return defaultHeight;
   }
 
   public static void applyKeyboardHeight(Context context, int height) {
+    SharedPreferences prefs = context.getSharedPreferences("KMAPreferences", Context.MODE_PRIVATE);
+    SharedPreferences.Editor editor = prefs.edit();
+    int orientation = getOrientation(context);
+    if (height == 0) {
+      // Passing 0 will reset both stored heights
+      editor.remove(KMManager.KMKey_KeyboardHeightLandscape);
+      editor.remove(KMManager.KMKey_KeyboardHeightPortrait);
+      editor.commit();
+
```

Comment:
```suggestion
      height = defaultHeight;
```

Reply:
I'm not sure if this one is still needed or where it goes. The variable is now defaultHeightForContext to distinguish it from the 0 static value.